doc/include/api_stdrt_cproc.inc

```diff
@@ -1,159 +1,106 @@
+the return due to reuse of the freed memory.
 
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%%}]
 [def Tcl_Obj*0]
 [def object0]
 
-The function returns a value of type [type Tcl_Obj*].
-
-This value becomes the interpreter result, if not [const 0]. The Tcl
-status is [const TCL_ERROR] when a [const 0] is returned, and
-[const TCL_OK] otherwise.
-
-[para][emph Note] how it is the responsibility of the function to set
-the interpreter result to an appropriate error message when returning
-[const 0].
-
-[para][emph Attention]: Contrary to [const Tcl_Obj*] above this
-conversion assumes that the returned value has a reference count of
-[const 0] and performs [emph no] decrement. Returning a value whose
-reference count is greater than zero will likely cause a memory leak.
+Like [const Tcl_Obj*] except that this conversion assumes that the
+returned value has a reference count of [const 0] and [emph {does not}]
+decrement it. Returning a value whose reference count is greater than zero is
+likely to cause a memory leak.
 
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%%}]
 [def char*]
 [def vstring]
 
-The function returns a value of type [type char*].
-
-This value becomes the interpreter result, wrapped in a [type String].
-
-It is assumed that the string is volatile in some way, with the
-wrapping in a [type String] duplicating it before making it the
-result, ensuring that we will not access a dangling pointer in the
-future.
+A [type String] Tcl_Obj holding a copy of the returned [type char*] is set as
+the interpreter result.
 
-The Tcl status is [emph always] [const TCL_OK].
 
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%%}]
 [def {const char*}]
 
-Like type [const char*] above, except that the returned string is
+Like [const char*] above, except that the returned string is
 [const const]-qualified.
 
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%%}]
 [def string]
 [def dstring]
 
-The function returns a value of type [type char*].
+The returned [type char*] is directly set as the interpreter result without
+making a copy.  Therefore it must be dynamically allocated via [fun Tcl_Alloc].
 
-Contrary to the previous string types here it is assumed that the
-value is dynamically allocated, via [fun Tcl_Alloc].
-
-This value becomes the interpreter result, as usual, but is [emph not]
-copied.
-
-The Tcl status is [emph always] [const TCL_OK].
 
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%%}]
 [def double]
-
-The function returns a value of type [type double].
-
-This value becomes the interpreter result, properly wrapped as a
-[type Double].
-
-The Tcl status is [emph always] [const TCL_OK].
-
-[comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%%}]
 [def float]
 
-The function returns a value of type [type float].
-
-This value becomes the interpreter result, properly wrapped as a
-[type Double].
+The returned [type double] or [type float] is converted [type Double] Tcl_Obj
```

Comment:
`converted` -> `converted to`

Reply:
or even `converted to a`, which is what the following types use

Reply:
Done with commit 29380521d0873ea1dcc5f8cdff6742b20764473f (Branch `documentation` in my repo, not seen by the PR).